### PR TITLE
DEVX-4979 DEVX-4980 Update dependencies

### PR DESCRIPTION
**Pull request by igorwojda**
Jira
https://nexmoinc.atlassian.net/browse/DEVX-4979
https://nexmoinc.atlassian.net/browse/DEVX-4980

Update dependencies and Gradle build scripts
- boost dependency versions
- migrate android-support to androidX
- extract AGP version to top-level `gradle.properties` file

This PR removes Travis CI, because it was crashing with the new Gradle plugin and it will be replaced with GitHub Actions in the next PR https://github.com/opentok/opentok-android-sdk-samples/pull/330 (review this PR first)

### CODE REVIEW

File: Archiving/app/build.gradle [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
+        minSdkVersion extMinSdkVersion
+        targetSdkVersion extTargetSdVersion
+        versionCode extVersionCode
+        versionName extVersionName
     }
 
     compileOptions {
-        sourceCompatibility sourcecompatibility
-        targetCompatibility targetcompatibility
+        sourceCompatibility JavaVersion.VERSION_1_8
```
Comment (rperezcubero) [inline Archiving/app/build.gradle:22]: Why aren't you also adding a `extSourceCompatibility` here?

File: Basic-Video-Chat/app/build.gradle [hunk 2]
```diff
@@ -1,34 +1,40 @@
-apply plugin: 'com.android.application'
-apply from: '../../build.properties'
+plugins {
+    id 'com.android.application'
+}
+
+apply {
+    from '../../build.properties'
+}
```
Comment (rperezcubero) [inline Basic-Video-Chat/app/build.gradle:7]: just curious but why are you changing these lines to the `{}` format?

**Comment by rperezcubero** [comment]
Left some comments but overall LGTM. Feel free to add the source compatibility to the root build.gradle file or just merge as it is.

**Review by rperezcubero — APPROVED** [review]

File: Basic-Video-Chat/app/build.gradle (hunk 2 above)
Comment (igorwojda) [reply Basic-Video-Chat/app/build.gradle:7]: I wanted to match the new Gradle format for applying plugins:

```
plugins {
    pluginA
}
```

but now when I am thinking about this - prev approach was better. 
I will revert this in following PRs

Comment (igorwojda) [reply Basic-Video-Chat/app/build.gradle:7]: Just to match the new Gradle format for applying plugins (that nowadays is used by AS when creating a new project):

image

Both will work

**Pull request merged**